backend/src/main/java/com/bakdata/conquery/models/worker/WorkerHandler.java

```diff
@@ -43,15 +49,40 @@ public IdMap<WorkerId, WorkerInformation> getWorkers() {
 		return this.workers;
 	}
 
+	public Map<UUID, PendingReaction> pendingReactions = new HashMap<>();
+
 	public void sendToAll(WorkerMessage msg) {
 		if (workers.isEmpty()) {
 			throw new IllegalStateException("There are no workers yet");
 		}
+
+		// Register tracker for pending reactions if applicable
+		if (msg instanceof ActionReactionMessage actionReactionMessage) {
+			final UUID callerId = actionReactionMessage.getMessageId();
+			pendingReactions.put(callerId, new PendingReaction(callerId, new HashSet<>(workers.keySet()), actionReactionMessage::afterAllReaction));
+		}
+
+		// Send message to all workers
 		for (WorkerInformation w : workers.values()) {
 			w.send(msg);
 		}
 	}
 
+	public void handleReactionMessage(ReactionMessage message) {
+		final UUID callerId = message.getCallerId();
+		final PendingReaction pendingReaction = pendingReactions.get(callerId);
+
+		if (pendingReaction == null) {
+			throw new IllegalStateException(String.format("No pending action registered (anymore) for caller id %s from reaction message: %s", callerId, message));
+		}
+
+		if (pendingReaction.checkoffWorker(message.getWorkerId())) {
+			log.debug("Removing pending reaction '{}' as last pending message was received.", callerId);
+			pendingReactions.remove(callerId);
```

Comment:
das wird auch in checkoff gemacht?

Reply:
Ich checkoff entfernen sich die "Worker" aus der pendingWorker liste.
Hier wird die pending Reaction komplett gelöscht